crates/provider-sqldb-postgres/src/bindings.rs

```diff
@@ -0,0 +1,1114 @@
+        ),
+        (top_right_x.integer_decode(), top_right_y.integer_decode()),
+    )
+}
+
+/// Convert a [`Linestring`] which represents a line/line segment and is expected to
+/// contain only *two* points, into a tuple of `HashableF64`s
+fn linestring_to_hashable_f64s_tuple(
+    l: LineString<f64>,
+) -> anyhow::Result<((HashableF64, HashableF64), (HashableF64, HashableF64))> {
+    match linestring_to_hashable_f64s(l)[..] {
+        [start, end] => Ok((start, end)),
+        _ => bail!("unexpected number of points in line string"),
+    }
+}
+
+/// Convert a [`Linestring`] into a vector of points (AKA two bindgen-generated `HashableF64`s)
+fn linestring_to_hashable_f64s(l: LineString<f64>) -> Vec<(HashableF64, HashableF64)> {
+    l.into_points()
+        .into_iter()
+        .map(point_to_hashable_f64s)
+        .collect::<Vec<_>>()
+}
+
+/// Convert a [`Point`] into two bindgen-generated `HashableF64`s
+fn point_to_hashable_f64s(p: Point<f64>) -> (HashableF64, HashableF64) {
+    let (x, y) = p.x_y();
+    (x.integer_decode(), y.integer_decode())
+}
+
+/// Convert a list of bytes to a MacAddress
+fn bytes_to_macaddr(bytes: impl AsRef<[u8]>) -> anyhow::Result<PgValue> {
+    let bytes = bytes.as_ref();
+    match bytes {
+        [octet0, octet1, octet2, octet3, octet4, octet5] => Ok(PgValue::Macaddr((
+            *octet0, *octet1, *octet2, *octet3, *octet4, *octet5,
+        ))),
+        _ => bail!("unexpected number of bytes in received macaddress"),
+    }
+}
+
+/// Convert a 6 byte MacAddress to a tuple of bytes
+fn macaddr_to_bytes(m: MacAddress) -> anyhow::Result<(u8, u8, u8, u8, u8, u8)> {
+    match m.as_bytes() {
+        [octet0, octet1, octet2, octet3, octet4, octet5] => {
+            Ok((*octet0, *octet1, *octet2, *octet3, *octet4, *octet5))
+        }
+        _ => bail!("unexpected number of bytes in received macaddress"),
+    }
+}
+
+impl TryFrom<MacAddress> for PgValue {
+    type Error = anyhow::Error;
+
+    fn try_from(m: MacAddress) -> anyhow::Result<PgValue> {
+        match m.as_bytes() {
+            bytes if bytes.len() == 8 => bytes_to_macaddr8(bytes),
+            bytes if bytes.len() == 6 => bytes_to_macaddr(bytes),
+            _ => bail!("unexpected number of bytes in macaddress"),
+        }
+    }
+}
+
+/// Convert a list of bytes to a MacAddress8
+fn bytes_to_macaddr8(bytes: impl AsRef<[u8]>) -> anyhow::Result<PgValue> {
+    let bytes = bytes.as_ref();
+    match bytes {
+        [octet0, octet1, octet2, octet3, octet4, octet5, octet6, octet7] => {
+            Ok(PgValue::Macaddr8((
+                *octet0, *octet1, *octet2, *octet3, *octet4, *octet5, *octet6, *octet7,
+            )))
+        }
+        _ => bail!("unexpected number of bytes in received macaddress"),
+    }
+}
+
+/// Convert a 6byte MacAddress to a tuple of bytes
+#[allow(clippy::type_complexity)]
+fn macaddr8_to_bytes(m: MacAddress) -> anyhow::Result<(u8, u8, u8, u8, u8, u8, u8, u8)> {
```

Comment:
Nit: Should this be an aliased type instead of a big tuple?

Reply:
Yeah, simple type alias

Reply:
Done! Since we had to remove `eui48` (it's got a vuln in a dep), this actually turned into it's own struct so we could hang implementations off of it.